filter_finalreport.py

```diff
@@ -0,0 +1,47 @@
+#! /usr/bin/env python3
+
+import argparse
+import glob
+
+
+def slice_file(input_file, chromosomes):
+    write_file = open("{}_sliced.txt".format(input_file[:-4]), "w")
+    input_lines = open(input_file, "r")
+    chrom_list = chromosomes.split(",")
+
+    """ Write all lines until Data section """
+    line = input_lines.readline()
+    while line.rstrip() != '[Data]':
+        write_file.write(line)
+        line = input_lines.readline()
+    write_file.write(line)  # Print [Data] line
+
+    """ Index header line for chromosome column, and write header"""
+    header = input_lines.readline()
+    chromosome_index = header.split("\t").index("Chr")
+    write_file.write(header)
+
+    """ Write all variant lines are on chromosomes of interest"""
+    for line in input_lines:
+        splitline = line.rstrip().split("\t")
+        if splitline[chromosome_index] in chrom_list:
+            write_file.write(line)
+
+    write_file.close()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--input_file', help='Full path to input file (must be a FinalReport.txt file)')
+    parser.add_argument(
+        '--chromosomes',
+        default="13,18,21,X,Y,XY",
```

Comment:
with `action=CheckExt({'txt'}),` you can also check if it is at least an .txt file and will automatically give a clear explanation what is wrong with the file. not needed, just something I knew about

Reply:
Extension of the argument input_file is actually not the most important. Should be a FinalReport file from Illumina Genomestudio, which are currently *FinalReport.txt.